cosmic-applet-time/src/window.rs

```diff
@@ -0,0 +1,328 @@
+                        y: y as i32,
+                        width: width as i32,
+                        height: height as i32,
+                    };
+                    get_popup(popup_settings)
+                }
+            }
+            Message::Tick => {
+                self.now = Local::now();
+                Command::none()
+            }
+            Message::Rectangle(u) => {
+                match u {
+                    RectangleUpdate::Rectangle(r) => {
+                        self.rectangle = r.1;
+                    }
+                    RectangleUpdate::Init(tracker) => {
+                        self.rectangle_tracker = Some(tracker);
+                    }
+                }
+                Command::none()
+            }
+            Message::CloseRequested(id) => {
+                if Some(id) == self.popup {
+                    self.popup = None;
+                }
+                Command::none()
+            }
+            Message::SelectDay(day) => {
+                // TODO
+                Command::none()
+            }
+            Message::PreviousMonth => {
+                // TODO
+                Command::none()
+            }
+            Message::NextMonth => {
+                // TODO
+                Command::none()
+            }
+            Message::OpenDateTimeSettings => {
+                // TODO
+                Command::none()
+            }
+        }
+    }
+
+    fn view(&self) -> Element<Message> {
+        let button = cosmic::widget::button(
+            if matches!(
+                self.core.applet.anchor,
+                PanelAnchor::Top | PanelAnchor::Bottom
+            ) {
+                column![
+                    cosmic::widget::text(self.now.format("%b %-d %-I:%M %p").to_string()).size(14)
+                ]
+            } else {
+                let mut date_time_col = column![
+                    icon::from_name("emoji-recent-symbolic")
+                        .size(self.core.applet.suggested_size().0)
+                        .symbolic(true),
+                    text(self.now.format("%I").to_string()).size(14),
+                    text(self.now.format("%M").to_string()).size(14),
+                    text(self.now.format("%p").to_string()).size(14),
+                    vertical_space(Length::Fixed(4.0)),
+                    // TODO better calendar icon?
+                    icon::from_name("calendar-go-today-symbolic")
+                        .size(self.core.applet.suggested_size().0)
+                        .symbolic(true),
+                ]
+                .align_items(Alignment::Center)
+                .spacing(4);
+                for d in self.now.format("%x").to_string().split("/") {
+                    date_time_col = date_time_col.push(text(d.to_string()).size(14));
+                }
+                date_time_col
+            },
+        )
+        .on_press(Message::TogglePopup)
+        .style(theme::Button::Text);
```

Comment:
After an update of libcosmic, this can be `.style(cosmic::theme::Button::AppletIcon)`